src/TGUI/Widgets/ChildWindow.cpp

```diff
@@ -396,6 +492,38 @@ namespace tgui
                 }
             }
 
+            // Check if the minimize button was clicked
+            else if (m_minimizeButton->m_mouseDown)
+            {
+                // Check if the mouse is still on the minimize button
+                if (m_minimizeButton->mouseOnWidget(x, y))
+                {
+                    if (isSignalBound("minimized"))
+                        sendSignal("minimized", std::static_pointer_cast<ChildWindow>(shared_from_this()));
+                    else
+                    {
+                        // TODO: Texus, should an exception be thrown or just ignore this case?
```

Comment:
You can just ignore the case because it just means that the user hasn't called `child->connect("minimized", ...)` yet. The buttons just don't do anything in this case.

I have noticed that the else statement can be removed entirely though (and so can the 'return' statement from several lines above for the close button) because there already seems to be a return statement directly after checking these buttons.